sdks/wasm/src/driver.c

```diff
@@ -674,3 +674,123 @@ mono_wasm_enable_on_demand_gc (void)
 {
 	mono_wasm_enable_gc = 1;
 }
+
+//
+// The following timezone information routines depends on the Mono_WebAssembly_ZoneInfo module being available.
+//
+
+// Returns the local timezone default is UTC.
+EM_JS(int, mono_wasm_timezone_get_local_name, (), {
+	var res = "UTC";
+	try {
+		return Mono_WebAssembly_ZoneInfo.mono_wasm_timezone_get_local_name ()
+	}
+	catch (wte) {
+		try { 
+			res = Intl.DateTimeFormat().resolvedOptions().timeZone; 
+		} 
+		catch(e) {} 
+
+		var buff = Module._malloc((res.length + 1) * 2);
+		stringToUTF16 (res, buff, (res.length + 1) * 2);
+		return buff;
+	}
+})
+
+void
+mono_timezone_get_local_name (MonoString *volatile*result)
+{
+	// WASM returns back an int pointer to a string UTF16 buffer.
+	// We then cast to `mono_unichar2*`.  Returning `mono_unichar2*` from the JavaScript call will
+	// result in cast warnings from the compiler.
+	mono_unichar2 *tzd_local_name = (mono_unichar2*)mono_wasm_timezone_get_local_name ();
+	*result = mono_string_from_utf16 (tzd_local_name);
+	free (tzd_local_name);
+}
+
+MonoString*
+mono_wasm_timezone_get_local_name_string (void)
+{
+	// WASM returns back an int pointer to a string UTF16 buffer.
+	// We then cast to `mono_unichar2*`.  Returning `mono_unichar2*` from the JavaScript call will
+	// result in cast warnings from the compiler.
+	mono_unichar2 *tzd_local_name = (mono_unichar2*)mono_wasm_timezone_get_local_name ();
+	MonoString *name = mono_string_from_utf16 (tzd_local_name);
+	free (tzd_local_name);
+	return name;
+}
+
+// Returns the timezone information for a specfic id.
+EM_JS(int, mono_wasm_zoneinfo_timezone_get_data, (int id, int *size), {
+
+	if (typeof Mono_WebAssembly_ZoneInfo !== "undefined")
+	{
+		var str = UTF8ToString (id);
+		return Mono_WebAssembly_ZoneInfo.mono_timezone_get_data(str, size);
+	}
+	return null;
+})
+
+void*
+mono_timezone_get_data (mono_unichar2 *name, int name_length, int *size)
+{
+
+	MonoString *zi_name;
+	if (!name)
+		zi_name = mono_wasm_timezone_get_local_name_string ();
+	else
+		zi_name = mono_string_new_utf16(root_domain, name, name_length);
+	// char *native_name = mono_string_to_utf8(zi_name);
+	// fprintf (stdout, "Called mono_timezone_get_data for %s - %i\n", native_name, name_length);
+	// mono_free (native_name);
+	*size = 0;
+	
+	char *native_val = mono_string_to_utf8 (zi_name);
+	int buffer = mono_wasm_zoneinfo_timezone_get_data((int)native_val, size);
+	return (void*)buffer;
```

Comment:
Do not cast pointers and possibly differently-sized integers. Use `gsize` to go between.
Better yet, use `gsize` here in place of `int`, for when wasm becomes 64bit.

Reply:
Feels like you leak `mono_string_to_utf8` but maybe not.

Reply:
Yes it is.  Thanks for the catch.

Reply:
leak fixed